services/QuillLMS/engines/comprehension/app/models/comprehension/rule.rb

```diff
@@ -98,5 +114,64 @@ def display_name
         errors.add(:prompts, "prompt #{prompt.id} already has a plagiarism rule") if prompt.rules.where(rule_type: TYPE_PLAGIARISM).first&.id
       end
     end
+
+    def log_creation(user_id)
+      if regex?
+        send_change_log(user_id, :create_regex, change_text_regex)
+      elsif plagiarism?
+        send_change_log(user_id, :create_plagiarism, change_text_plagiarism)
+      elsif universal?
+        send_change_log(user_id, :create_universal, change_text_universal)
+      end
+    end
+
+    def log_deletion(user_id)
+      if regex?
+        send_change_log(user_id, :delete_regex, nil, change_text_regex)
+      end
+    end
+
+    def log_update(user_id, old_values_arr, new_values_arr=nil)
+      if automl? && label.present? && !old_values_arr.select { |n| n.key?("name")}.empty?
+        prev_values = change_text_automl(old_values_arr.select { |n| n.key?("name")}.first["name"])
+        send_change_log(user_id, :update_semantic, "#{label.name} | #{name}", prev_values)
+      elsif regex?
+        prev_values = change_text_regex(old_values_arr)
+        send_change_log(user_id, :update_regex, change_text_regex(new_values(new_values_arr || old_values_arr)), prev_values)
+      elsif plagiarism?
+        prev_values = change_text_plagiarism(old_values_arr)
+        send_change_log(user_id, :update_plagiarism, change_text_plagiarism(new_values(new_values_arr || old_values_arr)), prev_values)
+      elsif universal?
+        prev_values = change_text_universal(old_values_arr)
+        send_change_log(user_id, :update_universal, change_text_universal(new_values(new_values_arr || old_values_arr)), prev_values)
+      end
+    end
+
+    private def send_change_log(user_id, action_type, new_value, prev_value=nil)
+      prompts&.each do |prompt|
```

Comment:
Again, I'm a bit worried about the potential duplication on Universal rules if that's not something we actively want.